independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanInfo.java

```diff
@@ -279,6 +279,15 @@ public List<InjectionPointInfo> getAllInjectionPoints() {
         return injectionPoints;
     }
 
+    boolean hasMetadataInjectionPoint() {
```

Comment:
I know naming this is hard, because the method is supposed to return whether this bean has an _injection point of type `InjectionPoint`_, but `hasMetadataInjectionPoint` sounds like the method returns whether this bean as an injection point for some metadata, which may be many things. How about `requiresInjectionPointMetadata`?

Reply:
`hasInjectionPointInjectionPoint`? ;-)
Jokes aside, I agree `requiresInjectionPointMetadata` is much better.

Reply:
:+1:

Reply:
I like `hasInjectionPointInjectionPoint` though :smile:

Reply:
No you don't :joy: